logging/proto_conversions_test.go

```diff
@@ -0,0 +1,128 @@
+package logging
+
+import (
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.viam.com/test"
+)
+
+func TestFieldConversion(t *testing.T) {
+	t.Parallel()
+
+	testTime, err := time.Parse(DefaultTimeFormatStr, "2024-03-01T11:39:40.897Z")
+	test.That(t, err, test.ShouldBeNil)
+
+	type s struct {
+		Field1 string
+		Field2 bool
+	}
+	testStruct := &s{"foo", true}
+
+	testCases := []struct {
+		field       zap.Field
+		expectedVal any
+	}{
+		{
+			field: zap.Field{
+				Key:     "boolean",
+				Type:    zapcore.BoolType,
+				Integer: 0,
+			},
+			expectedVal: false,
+		},
+		{
+			field: zap.Field{
+				Key:     "duration",
+				Type:    zapcore.DurationType,
+				Integer: time.Hour.Nanoseconds(),
+			},
+			expectedVal: time.Hour,
+		},
+		{
+			field: zap.Field{
+				Key:     "float64",
+				Type:    zapcore.Float64Type,
+				Integer: 4608218246714312622,
+			},
+			expectedVal: float64(1.23),
```

Comment:
Assuming these float tests passing in CI for amd+arm, they are good and fine by me. But just broadcasting that these assertions on floats are where my spidey senses go off where we might be relying on chip behavior.

Reply:
Yeah that's a fair worry; is it meaningful that this test passes in 32 bit architecture as well (according to CI)?

Reply:
In the context of CI, it's meaningful that the test doesn't fail, causing messages to spam #dev-ops and create JIRA tickets for netcode to fix!

If this was failing in CI on 32-bit, I suppose I'd want to make sure the production/non-testing code roughly works there, but not sure what our options would be if it didn't.